fix to content deletes

joelvarty opened the pull request:
Deleted or unpublished content was only not calling store.mergeItemToList so that it could be removed the respective List object that it was part of.

This has been fixed.

jamesvidler reviewed (commented):
@joelvarty it looks like the logic here is going to cause `store.mergeItemToList` to run twice for a deleted item. First it will `delete` the item, then it will call `mergeItemToList` where the store is now responsible to either delete or merge an item. Then, it flows into another if statement on line 130 where it will call `mergeItemToList` again.

I think you should be able to modify this so it is only called once.

joelvarty commented:
@jamesvidler Nice catch!  I've updated the delete logic to remove the duplicate merge call. I also updated the comments to be a little more descriptive.
